tuf/api/metadata.py

```diff
@@ -1299,3 +1299,34 @@ def to_dict(self) -> Dict[str, Any]:
     def update(self, fileinfo: TargetFile) -> None:
         """Assigns passed target file info to meta dict."""
         self.targets[fileinfo.path] = fileinfo
+
+    def add_key(self, role: str, key: Key) -> None:
+        """Adds new signing key for delegated role 'role'.
+
+        Raises:
+            KeyError: If there are no delegations, delegate role 'role' or key
+                with id `key.keyid`.
+        """
+        if self.delegations is None:
+            raise KeyError("Delegations doesn't exists!")
+        self.delegations.roles[role].keyids.add(key.keyid)
+        self.delegations.keys[key.keyid] = key
+
+    def remove_key(self, role: str, keyid: str) -> None:
+        """Removes key from delegated role 'role' and updates the delegations
+        key store.
+
+        Raises:
+            ValueError: If there are no delegated roles or if 'role' is not
```

Comment:
How did we end up with two equivalent functions throwing `ValueError` and `KeyError` for the same thing. I am not sure about this.

Reply:
Now as you highlighted it, it's not great we throw two different types of exceptions for `Root.remove_key()` - `KeyError` and `Targets.remove_key()` - `ValueError`.
IMO the correct exception here should be `ValueError` as the problems are in the given **values** -`role` doesn't exist or `keyid` doesn't point to a key used by that role.
@jku and @sechkova if you both agree I can change the `Targets.remove_key()` exception to be `ValueError` as well.

Reply:
I suppose you mean _change Targets.add_key() exception_: sounds good to me. ValueError is no doubt the correct error (but it mostly matters for aesthetics: this is a programmer error not an error that will be handled)

btw: you don't need to say _"If there are no delegated roles"_ in the docstring: that is an implementation detail the caller does not need to know: for them the error is _"'role' is not delegated by this Targets"_

Reply:
I made an API change to `Root.add_key()` and `Root.remove_key()` so that they throw `ValueError` instead of `KeyError` on wrong given arguments.
The reason for this change is that `ValueError` in my opinion is the right exception that should be thrown when the value is not correct and I think it's important to be consistent when calling `delegator.add_key` or delegator.remove_key` no matter if the delegator is `Root` or `Targets`.